src/Analysis/SharedUtils/WorkflowLimitsDashboard/WorkflowLimitsUtils.js

```diff
@@ -0,0 +1,31 @@
+import { headers } from '../../../configs';
+import { gwasWorkflowPath } from '../../../localconf';
+
+export const workflowLimitsLoadingErrorMessage = 'Unable to gather user workflow information.';
+
+export const workflowLimitsInvalidDataMessage = 'Invalid server response for user workflow information.';
+
+export const fetchMonthlyWorkflowLimitInfo = async () => {
+  const workflowsEndpoint = `${gwasWorkflowPath}workflows/user-monthly`;
+  const response = await fetch(workflowsEndpoint, { headers });
+  if (!response.ok) {
+    const message = `An error has occurred: ${response.status}`;
+    throw new Error(message);
+  }
+  return response.json();
+};
+
+export const workflowLimitInfoIsValid = (data) => {
+  // Check if data is an object
+  if (typeof data !== 'object' || data === null) {
+    return false;
+  }
+  // validate data contains expected keys and they're numeric
+  if (
+    typeof data?.workflow_run !== 'number'
+    || typeof data?.workflow_limit !== 'number'
+  ) {
+    return false;
+  }
+  return true;
+};
```

Comment:
because of `workflowRun / workflowLimit` done in other part of the code, I think this function should also return `false` if `workflowLimit === 0`

Reply:
Good catch, function updated along with unit tests for this edge case. 
```
  // validate data contains expected keys and they're numeric
  // and workflow limit is greater than 0
  if (
    typeof data?.workflow_run !== 'number'
    || typeof data?.workflow_limit !== 'number'
    || !(data.workflow_limit > 0)
  ) {
    return false;
  }
  ```